draft-ietf-oauth-jwk-thumbprint-uri.xml

```diff
@@ -129,6 +129,32 @@
         The security considerations of <xref target="RFC7638"/>
 	also apply when using this specification.
       </t>
+
+      <section anchor="MultiplePublicKeysPerPrivateKey" title="Multiple Public Keys per Private Key">
+	<t>
+	  There are cryptographic algorithms for which multiple public keys correspond to the same private key.
+	  This is described in the security considerations of <xref target="RFC7748"/> as follows:
+	</t>
+	<t>
+	  <list style="empty">
+	    <t>
+	      Designers using these curves should be aware that for each public
+	      key, there are several publicly computable public keys that are
+	      equivalent to it, i.e., they produce the same shared secrets.  Thus
+	      using a public key as an identifier and knowledge of a shared secret
+	      as proof of ownership (without including the public keys in the key
+	      derivation) might lead to subtle vulnerabilities.
```

Comment:
I think "key derivation" should be clarified a little more.

Reply:
I actually think that the RFC authors misspoke by using the term "key derivation" above.  I believe they are actually talking about including the public key in the derivation or transmission of the identifier used.  That said, I could be wrong and if so, I hope Neil will point out how the current security considerations text should be improved.

We're accomplishing this as described in the following paragraph "A recommended way to ensure that the JWK Thumbprint URI corresponds to the actual public key used is to sign a message containing the correct public key with the private key."

If you have other specific wording improvements in mind, let me know.

Reply:
Interesting. I will think about this..